x-pack/plugins/infra/public/pages/metrics/hosts/components/host_details_flyout/metadata/add_metadata_filter_button.tsx

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiToolTip, EuiButtonIcon } from '@elastic/eui';
+import { Filter, PhraseFilter, PhrasesFilter, ScriptedPhraseFilter } from '@kbn/es-query';
+import { buildMetadataFilter } from './build_metadata_filter';
+import { useMetricsDataViewContext } from '../../../hooks/use_data_view';
+import { useUnifiedSearchContext } from '../../../hooks/use_unified_search';
+import { useKibanaContextForPlugin } from '../../../../../../hooks/use_kibana';
+
+interface AddMetadataFilterButtonProps {
+  item: {
+    name: string;
+    value: string | string[] | undefined;
+  };
+}
+
+const filterAddedToastTitle = i18n.translate(
+  'xpack.infra.hostsViewPage.flyout.metadata.filterAdded',
+  {
+    defaultMessage: 'Filter was added',
+  }
+);
+
+export const AddMetadataFilterButton = ({ item }: AddMetadataFilterButtonProps) => {
+  const { dataView } = useMetricsDataViewContext();
+  const { searchCriteria } = useUnifiedSearchContext();
+  const [existingFilter, setExistingFilter] = useState<
+    PhrasesFilter | PhraseFilter | ScriptedPhraseFilter | Filter | undefined
+  >(() => undefined);
```

Comment:
**nit**: initializing to undefined is not an expensive task, you can pass it directly without the need for a lazy init state.

Reply:
After the change proposed later, this state should be removed.

Reply:
Thank you for the suggestions here - I was thinking to build the filter here as a default state but after that decided that it's not needed because it's used only for adding filters and forgot to update it. I applied the suggested changes 👍